bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -0,0 +1,387 @@
+                        logger.trace("Consumption at index {} added to current week: {}", jump, consumption);
+                    }
+                    yesterday = consumption;
+                }
+                jump++;
+                rangeStart = rangeStart.plusDays(1);
+            }
+        } else {
+            cachedDaylyData.invalidateValue();
+        }
+        updateKwhChannel(YESTERDAY, yesterday);
+        updateKwhChannel(THIS_WEEK, thisWeek);
+        updateKwhChannel(LAST_WEEK, lastWeek);
+    }
+
+    /**
+     * Request new monthly data and updates channels
+     */
+    private synchronized void updateMonthlyData() {
+        if (!isLinked(LAST_MONTH) && !isLinked(THIS_MONTH)) {
+            return;
+        }
+
+        double lastMonth = -1;
+        double thisMonth = -1;
+        LinkyConsumptionData result = cachedMonthlyData.getValue();
+        if (result != null && result.success()) {
+            int jump = result.getDecalage();
+            lastMonth = result.getData().get(jump++).valeur;
+            thisMonth = result.getData().get(jump).valeur;
+            if (thisMonth < 0) {
+                thisMonth = 0;
+            }
+        } else {
+            cachedMonthlyData.invalidateValue();
+        }
+        updateKwhChannel(LAST_MONTH, lastMonth);
+        updateKwhChannel(THIS_MONTH, thisMonth);
+    }
+
+    /**
+     * Request new yearly data and updates channels
+     */
+    private synchronized void updateYearlyData() {
+        if (!isLinked(LAST_YEAR) && !isLinked(THIS_YEAR)) {
+            return;
+        }
+
+        double thisYear = -1;
+        double lastYear = -1;
+        LinkyConsumptionData result = cachedYearlyData.getValue();
+        if (result != null && result.success()) {
+            int elementQuantity = result.getData().size();
+            thisYear = elementQuantity > 0 ? result.getData().get(elementQuantity - 1).valeur : -1;
+            lastYear = elementQuantity > 1 ? result.getData().get(elementQuantity - 2).valeur : -1;
+        } else {
+            cachedYearlyData.invalidateValue();
+        }
+        updateKwhChannel(LAST_YEAR, lastYear);
+        updateKwhChannel(THIS_YEAR, thisYear);
+    }
+
+    private void updateKwhChannel(String channelId, double consumption) {
+        logger.debug("Update channel {} with {}", channelId, consumption);
+        updateState(channelId,
+                consumption != -1 ? new QuantityType<>(consumption, SmartHomeUnits.KILOWATT_HOUR) : UnDefType.UNDEF);
+    }
+
+    /**
+     * Produce a report of all daily values between two dates
+     *
+     * @param startDay the start day of the report
+     * @param endDay the end day of the report
+     * @param separator the separator to be used betwwen the date and the value
+     *
+     * @return the report as a string
+     */
+    public String reportValues(LocalDate startDay, LocalDate endDay, @Nullable String separator) {
+        String dump = "";
```

Comment:
Please use StringBuilder instead of string concatenation to build up this value.